Task 2/Task 2.1/CustomString/Class1.cs

```diff
@@ -0,0 +1,187 @@
+                        flag = true;
+                    }
+                }
+
+                return flag;
+            }
+        }
+
+        /// <summary>
+        /// Метод, определяющий, есть ли искомый символ в кастомной строке.
+        /// </summary>
+        /// <param name="Symbol"></param>
+        /// <returns>Возвращает true или false, в зависимости от наличия символа в строке.</returns>
+        public bool SearchSymbol(char Symbol)
+        {
+            bool flag = false;
+            foreach (char item in MyString)
+            {
+                if (item == Symbol)
+                    flag = true;
+            }
+            return flag;
+        }
+
+        /// <summary>
+        /// Метод, подсчитывающий количество искомых символов в кастомной строке.
+        /// </summary>
+        /// <param name="Symbol"></param>
+        /// <returns>Возвращает количество символов.</returns>
+        public int CountSymbol(char Symbol)
+        {
+            int count = 0;
+
+            foreach (char item in MyString)
+            {
+                if (item == Symbol) count++;
+            }
+
+            return count;
+        }
+
+        public void Print()
+        {
+            foreach (char item in MyString)
+            {
+                Console.Write(item);
+            }
+        }
+
+        /// <summary>
+        /// Метод, возвращающий строку, состоящую из символов, хранящихся в кастомной строке,
+        /// собранных в порядке справа-налево.
+        /// </summary>
+        /// <returns></returns>
+        public string Reverse()
+        {
+            string reverse = string.Empty;
+
+            for (int i = MyString.Length - 1; i >= 0; i--)
+            {
+                reverse += MyString[i];
+            }
+
+            return reverse;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as CustomString);
+        }
+
+        public bool Equals(CustomString other)
+        {
+            return other != null &&
+                   EqualityComparer<char[]>.Default.Equals(_myString, other._myString) &&
+                   EqualityComparer<char[]>.Default.Equals(MyString, other.MyString);
+        }
+
+        public override int GetHashCode()
+        {
+            return HashCode.Combine(_myString, MyString);
+        }
+
+        public char this[int index]
+        {
+            get
+            {
+                if (index > MyString.Length - 1 || index < 0)
+                {
+                    return ' ';
+                }
+                else
+                {
+                    return MyString[index];
+                }
+            }
+            set
+            {
+                if (index > MyString.Length - 1 || index < 0)
+                {
+                    Console.WriteLine("Такого индекса не существует! Сначала инициализируйте строку необходимой длины для того, чтобы что-то менять.");
+                }
```

Comment:
Writing to console from inside the class to say something to user is a bad idea. For your case class should be throwing an exception. (Some additional data can be printed to log/Debug.WriteLine though, but that's another story)

```csharp
throw new IndexOutOfRangeException("message");
```